h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -146,7 +148,13 @@ private static int ensureBuilderIndex(String urlName) {
    *  Shallow clone of both the default ModelBuilder instance and a Parameter. */
   public static <B extends ModelBuilder> B make(String algo, Job job, Key<Model> result) {
     int idx = ArrayUtils.find(ALGOBASES,algo.toLowerCase());
-    assert idx != -1 : "Unregistered algorithm "+algo;
+    if (idx < 0) {
+      StringBuilder sb = new StringBuilder();
+      sb.append("Unknown algo: ").append(algo).append("Extension report: ");
+      Log.err(ExtensionManager.getInstance().makeExtensionReport(sb));
```

Comment:
I really like the concept of extension report, however do you think it's helpful to report it in this case? Algorithms are independent on the extensions. 

On the other hand, algorithms are independent on the extensions at this moment, in the future we might add algos as extensions and this becomes handy

Reply:
The issue of PUBDEV-5290 is xgboost not being available. XGBoost is itself an extension (core extension). The idea is to capture the state of extensions at the moment the algo (=xgboost) was not found. I don't want to rely on the prior logs (older state) because it might not be consistent with the current state.

The idea of this change is to get the picture what was going on at the time the algo was not found. Does that make sense?

Reply:
Makes sense, great explanation, thanks!